core/src/main/java/uk/gov/justice/services/core/jms/AbstractJMSListener.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.services.core.jms;
+
+
+import uk.gov.justice.services.core.dispatcher.Dispatcher;
+import uk.gov.justice.services.core.jms.converter.EnvelopeConverter;
+import uk.gov.justice.services.core.jms.exception.InvalildJmsMessageTypeException;
+
+import javax.inject.Inject;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageListener;
+import javax.jms.TextMessage;
+
+/**
+ * Abstract JMS Listener. Dispatches the message to the correct handler.
+ */
+public abstract class AbstractJMSListener implements MessageListener {
+
+    @Inject
+    EnvelopeConverter envelopeConverter;
+
+    protected abstract Dispatcher getDispatcher();
+
+    @Override
+    public void onMessage(final Message message) {
+        if (!(message instanceof TextMessage)) {
+            try {
+                throw new InvalildJmsMessageTypeException("Message is not an instance of TextMessage " + message.getJMSMessageID());
```

Comment:
Possibly, String "Message is not an instance of TextMessage" could be made a constant.